City-of-Helsinki/drupal-helfi-rekry
# UHF-5291 listing translations

# [UHF-5291](https://helsinkisolutionoffice.atlassian.net/browse/UHF-5291)
Handle unused translations.

## What was done
* Added event subscriber and Queue Worker to handle untranslated job listings
* Added logic to wrap untranslated node fields with `lang` attribute

## How to install

* Make sure your instance is up and running on correct branch.
  * `git fetch`
  * `git checkout origin/UHF-5921-listing-translations`
  * `make build drush-cr drush-cim`
* Run `make drush-cr`

## How to test
* Add `HELBIT_CLIENT_ID` to your .env (correct value in Jira ticket)
  * Run `make up`. Your app container should restart
  * Run `make shell`
  * Run `printenv | grep HELBIT_CLIENT_ID` in the container. This variable should have value
* Run `drush mim helfi_rekry_jobs:all --limit=100 --update`
* Run `drush queue:list`
  * `helfi_rekry_job_translations` should have items queued
* Run `drush queue:run helfi_rekry_job_translations`
  * If you run `drush queue:list` again, the queue should be empty.
* Go to content page https://helfi-rekry.docker.so/admin/content
* Check the translations for the most recent job listing node.
  * At least one of the translations should have `field_copied` value as `TRUE`
  * Ones with this value marked should also have `field_original_language` value
  * View the node. Untranslated field values should be wrapped with element with HTML attribute `lang` and appropriate value

## Designers review
